contrib/remote-ssh-build.bash

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+## REMOTE SSH ACCESS
+
+# This script builds a zbm with remote access support via ssh. The image and
+# efi executable will contain private host keys and public keys used for
+# mutual authentication.
+
+# All keys / settings are placed into ${RS_DIR} ('ssh-data' in the current
+# working directory by default). Its contents can be customized as needed.
+# They are regenerated if missing.
+
+# Server private keys are generated automatically as needed. It is
+# recommended to not reuse existing private keys.
+# User authentication for the root account is done via 'authorized_keys'.
+# The current users' authorized keys are copied if it is not present.
+
+RS_DIR=$(realpath "${RS_DIR:-ssh-data}")
+RS_KEYDIR="${RS_DIR}/keys"
+
+mkdir -p "${RS_DIR}"
+mkdir -p "${RS_KEYDIR}"
+
+
+## PREPARE SERVER PRIVATE KEYS
+
+# If no local ssh host keys are available, create them
+# WARNING: These private keys will be part of the final image!
+RS_PK_RSA="${RS_KEYDIR}"/ssh_host_rsa_key
+RS_PK_ECDSA="${RS_KEYDIR}"/ssh_host_ecdsa_key
+RS_KEYGEN_OPT=(-m PEM -N '' -C "zbm-server-key")
+if [ ! -f "${RS_PK_RSA}" ]; then
+  ssh-keygen -t rsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_RSA}"
+fi
+if [ ! -f "${RS_PK_ECDSA}" ]; then
+  ssh-keygen -t ecdsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_ECDSA}"
+fi
+
+
+## PREPARE AUTHORIZED KEYS
+
+RS_AUTH="${RS_KEYDIR}/authorized_keys"
+if [ ! -f "${RS_AUTH}" ]; then
+  echo "Cannot find ${RS_AUTH}, copying from current user"
+  cp -v ~/.ssh/authorized_keys "${RS_AUTH}"
+fi
+
+
+## PREPARE SETTINGS
+
+RS_DNC="${RS_DIR}/dracut-network.conf"
+if [ ! -f "${RS_DNC}" ]; then
+  # ip=dhcp tries to bring up all interfaces
+  # ip=single-dhcp stops after bringing up the first
+  # See https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html
+  echo "ip=single-dhcp rd.neednet=1" > "${RS_DNC}"
+fi
+
+RS_DDC="${RS_DIR}/dracut-dropbear.conf"
+if [ ! -f "${RS_DDC}" ]; then
+  cat >> "${RS_DDC}" <<EOF
+# Enable dropbear ssh server and pull in network configuration args
+add_dracutmodules+=" crypt-ssh "
+install_optional_items+=" /etc/cmdline.d/dracut-network.conf "
+dropbear_rsa_key=/etc/dropbear/ssh_host_rsa_key
+dropbear_ecdsa_key=/etc/dropbear/ssh_host_ecdsa_key
+dropbear_acl=/etc/dropbear/authorized_keys
+EOF
+fi
```

Comment:
Just dump this right in `dracut.conf.d`, and why bother mounting stuff into `/etc/dropbear` when `./dropbear` is visible inside the container as `/build/dropbear`? We can point right to that.

Also, add tabs to make things more readable. (They have to be hard tabs, not the two spaces added by the modeline, so `Ctrl-v <tab>` it is.
```suggestion
mkdir -p dracut.conf.d
RS_DDC="${RS_DIR}/dracut.conf.d/dracut-dropbear.conf"
if [ ! -f "${RS_DDC}" ]; then
  cat >> "${RS_DDC}" <<-EOF
	# Enable dropbear ssh server and pull in network configuration args
	add_dracutmodules+=" crypt-ssh "
	install_optional_items+=" /etc/cmdline.d/dracut-network.conf "
	dropbear_rsa_key=/build/dropbear/ssh_host_rsa_key
	dropbear_ecdsa_key=/build/dropbear/ssh_host_ecdsa_key
	dropbear_acl=/build/dropbear/authorized_keys
	EOF
fi
```

Reply:
This won't work as-is, since `${RS_DIR}` is not usually the current working directory. I either have to ignore `${RS_DIR}`, and dump some of the configuration into the current working directory (the default build directory), or maybe change the build directory to `$RS_DIR`.